### ISSUE TITLE
about mmdeteciton3d version

### ISSUE THREAD

**Issue by zyrant** [Haiyang-W/CAGroup3D#5]
Hi, I need to do data preprocessing by myself, but as mentioned in https://github.com/SamsungLabs/fcaf3d/issues/42#issuecomment-1242940335, different versions of mmdeteciton3d may cause different preprocessing outputs. what version of this report is used?

**Comment by Haiyang-W** [comment]
We provide the processed data in https://drive.google.com/drive/folders/1sKvq4WBSEb4CWMdCTN6lCHLXnn3NwUv_  . As for the version, we follow the FCAF3D and use mmdet3d-0.15.0. Hope that can ease your concerns.

**Comment by zyrant** [comment]
Thanks.  Sorry for bother again.
Since there is no pre-training weight provided at present, I retrained the model with 8*2, but it exceeded the GPU memory limit, I could only use 4*2 and got the following result.
[log_train_20230220-205910.txt](https://github.com/Haiyang-W/CAGroup3D/files/10793130/log_train_20230220-205910.txt)
In addition, considering a series of problems caused by bn, etc., I also tried to adjust _lr_ to 0.0005 and got the following result.
[log_train_20230221-094207.txt](https://github.com/Haiyang-W/CAGroup3D/files/10793151/log_train_20230221-094207.txt)
Although the result is the best level at present, it is still a certain distance from the result in the paper.

**Comment by pigtigger** [comment]
Here is one of our pcdet training logs for ScanNetV2.
[log_train_20230131-234426.txt](https://github.com/Haiyang-W/CAGroup3D/files/10800489/log_train_20230131-234426.txt)
Your low performance may be due to the batch size, which may influence the BN layers and lead to unsatisfactory results.  Could you use 8x2  or 4x4 to retrain the model? (Maybe you could resume the checkpoint If you exceed the GPU memory during an epoch)
Btw, there are certain fluctuations in performance on ScanNetv2, did you fix the random seed by adding --fix_random_seed when training?

**Comment by Haiyang-W** [comment]
> Thanks. Sorry for bother again. Since there is no pre-training weight provided at present, I retrained the model with 8×2, but it exceeded the GPU memory limit, I could only use 4×2 and got the following result. [log_train_20230220-205910.txt](https://github.com/Haiyang-W/CAGroup3D/files/10793130/log_train_20230220-205910.txt) In addition, considering a series of problems caused by bn, etc., I also tried to adjust lr to 0.0005 and got the following result. [log_train_20230221-094207.txt](https://github.com/Haiyang-W/CAGroup3D/files/10793151/log_train_20230221-094207.txt) Although the result is the best level at present, it is still a certain distance from the result in the paper.

We have reproduced the CAGroup3D on OpenPCDet (may be slightly lower than mmdet3d version in mAP@0.25: 74.5->74.1 in average and higher in mAP@0.5: 60.3->60.9). The mmdet3d version will be released in few weeks, because i am rushing some DDLs. I'll also try my best to fix this gain (0.4) on mAP@0.25 between pcdet and mmdet3d. Hope that can solve your concerns. 

Best regards,
Haiyang

**Comment by zyrant** [comment]
Thanks. @pigtigger @Haiyang-W 
After reading your answers. I don't think it's a problem anymore.

**Comment by Haiyang-W** [comment]
> Thanks. @pigtigger @Haiyang-W After reading your answers. I don't think it's a problem anymore.

Great! If you have any questions, please feel free to contact me. :)

Have a good day!
Haiyang

**Comment by zyrant** [comment]
你好
最近，我尝试了个采样10w个点的CAGroup的版本，结果十分奇怪。
下面是我的训练日志和yaml文件。
[log_train_20230304-183320.txt](https://github.com/Haiyang-W/CAGroup3D/files/10899351/log_train_20230304-183320.txt)
[CAGroup3D.txt](https://github.com/Haiyang-W/CAGroup3D/files/10899355/CAGroup3D.txt)

我这个问题不急，祝你在ICCV DDL前完成你的工作，等到DDL后在回复即可。

顺便恭喜Dynamic Sparse Voxel Transformer with Rotated Sets被CVPR 2023接受了！

祝好

**Comment by Haiyang-W** [comment]
> 你好 最近，我尝试了个采样10w个点的CAGroup的版本，结果十分奇怪。 下面是我的训练日志和yaml文件。 [log_train_20230304-183320.txt](https://github.com/Haiyang-W/CAGroup3D/files/10899351/log_train_20230304-183320.txt) [CAGroup3D.txt](https://github.com/Haiyang-W/CAGroup3D/files/10899355/CAGroup3D.txt)
> 
> 我这个问题不急，祝你在ICCV DDL前完成你的工作，等到DDL后在回复即可。
> 
> 顺便恭喜Dynamic Sparse Voxel Transformer with Rotated Sets被CVPR 2023接受了！
> 
> 祝好

谢谢您～，这个问题我觉得可能是indoor sample这个data augmentation您那边没用对？或许你可以参考一下RBGNet里关于indoor_point_sample的用法，[indoor_point_sample_rbgnet](https://github.com/Haiyang-W/RBGNet/blob/c57da64740e4e7ba4ea31bad70b41ccc5791ddbc/tools/cfgs/dataset_configs/scannet_dataset.yaml#L38)。另外也要注意这个config里的DATA_AUGMENTOR和DATA_AUGMENTOR_TEST也要相应改动。

祝好！
海洋

**Comment by zyrant** [comment]
是的是的，您是对的，是confg文件出错了。
我以为DATA_AUGMENTOR_TRAIN和DATA_AUGMENTOR_TEST会继承_BASE_CONFIG_的相关参数，其实是覆写。我把scannet_dataset.yaml中的相关参数全都复制到了CAGroup3D.yaml中，重新跑了一轮，结果很正常了。

感谢您耐心的解答。
祝好！

**Comment by Haiyang-W** [comment]
> 是的是的，您是对的，是confg文件出错了。 我以为DATA_AUGMENTOR_TRAIN和DATA_AUGMENTOR_TEST会继承_BASE_CONFIG_的相关参数，其实是覆写。我把scannet_dataset.yaml中的相关参数全都复制到了CAGroup3D.yaml中，重新跑了一轮，结果很正常了。
> 
> 感谢您耐心的解答。 祝好！

解决就好～，PCDet对于在test的时候用da有点麻烦，所以这里为了少改动写的不好。希望一切顺利! :)
海洋

**Comment by zyrant** [comment]
hello，抱歉又打扰了。
我最近想把roi head搬到mmdet3d下，不过遇到问题不少，不知道你们有计划公开mmdet3d的代码吗？或者能发一份基于mmdet3d的代码到我的邮箱吗(zhu.yun@njust.edu.cn)？感谢！

祝好

**Comment by Haiyang-W** [comment]
> hello，抱歉又打扰了。 我最近想把roi head搬到mmdet3d下，不过遇到问题不少，不知道你们有计划公开mmdet3d的代码吗？或者能发一份基于mmdet3d的代码到我的邮箱吗([zhu.yun@njust.edu.cn](mailto:zhu.yun@njust.edu.cn))？感谢！
> 
> 祝好

I have sent it to you by email. If you have any questions, please feel free to contact me.

Haiyang,

**Comment by zyrant** [comment]
thanks.

best wishes.
